rasterio/crs.pyx

```diff
@@ -924,27 +924,19 @@ cdef class CRS:
         cdef OGRSpatialReferenceH osr_s = NULL
         cdef OGRSpatialReferenceH osr_o = NULL
         cdef CRS crs_o
-
+        cdef const char* options[2]
+        options[0] = b"IGNORE_DATA_AXIS_TO_SRS_AXIS_MAPPING=YES"
+        options[1] = NULL
         try:
             crs_o = CRS.from_user_input(other)
+            osr_s = exc_wrap_pointer(OSRClone(self._osr))
+            osr_o = exc_wrap_pointer(OSRClone(crs_o._osr))
+            return bool(OSRIsSameEx(osr_s, osr_o, options) == 1)
```

Comment:
maybe I'm missing something but why not doing directly 
```
self._osr.IsSame(crs_o._osr, options)
```

Reply:
I'm not 100% sure but I think that IsSame is only a function of the `OGRSpatialReference` c++ class rather than the `OGRSpatialReferenceH` type. It doesn't look like that class is wrapped in cython yet. Would probably require replacing the ._osr with a `OGRSpatialReference` object and could get messy

Reply:
You could likely save the OSRClone() and just do:
```
return bool(OSRIsSameEx(self._osr, crs_o._osr, options) == 1)
```

Reply:
about to push but to update the micro benchmarks I posted before:
```python
In [5]: %timeit e.wkt == e.wkt
46.3 ns ± 0.804 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [6]: %timeit e == e
242 ns ± 15.6 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [7]: %timeit m == m
236 ns ± 3.42 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [8]: m2 = rio.CRS.from_wkt('GEOGCS["Mars (2015) - Sphere XY",DATUM["Mars (2015) - Sphere",SPHEROID["Mars (2015) -Sphere",3396190,0]],PRIMEM["Reference Meridian",0],UNIT["degree",0.01745329
   ...: 25199433,AUTHORITY["EPSG","9122"]],AXIS["Longitude",EAST],AXIS["Latitude",NORTH]]')
   ...: 

In [9]:  m2 == m
Out[9]: True

In [10]: %timeit m == m2
4 µs ± 16.3 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

```